### DISCUSSION TITLE
How should I copy the RefArch repo to our Monorepo?

### DISCUSSION THREAD

**Discussion by vicyap** [gruntwork-io/knowledge-base#435] [Help] [answered]
Hi, I just want to sanity check. We just got our RefArch deployed and want to move it into our Monorepo. We want to do this so we can make infrastructure changes and application changes in a single PR.

After reading this: https://github.com/gruntwork-io/knowledge-base/discussions/227, is it as simple as copying everything in `infrastructure-live/` to `our-monorepo/infra/live/`? Then we can use `our-monorepo/infra/modules` for any terraform modules we may eventually implement?

---

<ins datetime="2022-05-25T16:46:47Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108660">Tracked in ticket #108660</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
The easiest would be to create a new folder in your existing mono repo and put everything in `infrastructure-live` into there. That way, you can ensure that all the necessary files that the project finds in the tree will terminate at that folder level instead of going further up the tree to your mono repo.

If you are looking to integrate the Reference Architecture into an existing `infrastructure-live` project, then the recommendation would be to identify the differences between your existing `infrastructure-live` root `terragrunt.hcl` config and to try and adapt your existing modules with the root `terragrunt.hcl` from the Reference Architecture. That way, you aren't working with multiple versions of the root `terragrunt.hcl` config or folder structure.
